exonum-java-binding-core/rust/integration_tests/tests/service_proxy.rs

```diff
@@ -194,3 +209,91 @@ fn service_can_modify_db_on_initialize() {
         .expect("Failed to find the entry created in the test service");
     assert_eq!(INITIAL_ENTRY_VALUE, value);
 }
+
+#[test]
+#[should_panic(expected = "Java exception: com.exonum.binding.fakes.mocks.TestException")]
+fn after_commit_throwing() {
+    let service = ServiceMockBuilder::new(EXECUTOR.clone())
+        .state_hashes(&[hash(&[1, 2, 3])])
+        .after_commit_throwing(TEST_EXCEPTION_CLASS)
+        .build();
+
+    // It turned out that it is MUCH easier to use testkit in order to trigger the after_commit()
+    // callback than calling it by hands providing manually constructed ServiceContext entity.
+    let mut testkit = TestKitBuilder::validator()
+        .with_service(service.clone())
+        .create();
+
+    testkit.create_block();
+}
+
+#[test]
+fn after_commit_validator() {
+    let (builder, interactor) = ServiceMockBuilder::new(EXECUTOR.clone())
+        .state_hashes(&[hash(&[1, 2, 3])])
+        .get_mock_interaction_after_commit();
+
+    let service = builder.build();
+    let mut testkit = TestKitBuilder::validator()
+        .with_service(service.clone())
+        .create();
+
+    testkit.create_block();
+    testkit.create_block();
+
+    let result = get_mock_interaction_result(&EXECUTOR, interactor.as_obj());
+    let res_iter: Vec<AfterCommitArgs> = from_str(&result).unwrap();
+
+    assert_eq!(res_iter.len(), 2);
+
+    let item: &AfterCommitArgs = res_iter.get(0).unwrap();
+    assert!(item.handle > 0);
```

Comment:
I wonder if handle can be negative 🤔 If it's pointer representation, it can, I suppose

Reply:
Most probably you're right... Then it makes sense to check for equality to zero only.